### ISSUE TITLE
Privacy plugin causes a 404 on css

### ISSUE THREAD

**Issue by dr-br** [squidfunk/mkdocs-material#7672]
### Bug description

Starting with 9.5.43 the privacy plugin causes an invalid include of a css file.  
The file `assets/external/fonts.googleapis.com/css.49ea35f2.css` is not found, resulting in a 404.  
The default Roboto font is not available, resulting in some less fancy backup font.

Version 9.5.42 and before is fine.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
-

### Reproduction

[9.5.43-privacy-plugin-bug.zip](https://github.com/user-attachments/files/17631982/9.5.43-privacy-plugin-bug.zip)

### Steps to reproduce

- Unzip the reproduction
- `mkdocs serve`
- comment/uncomment the privacy plugin results in different font styles and a 404 in the dev tools of my browser

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Tagging @nejch for visibility – there seems to be some side effect in the merged changed. The CSS file is downloaded but not copied to the download directory.

**Comment by nejch** [comment]
That does seem to be the case, e.g. 

https://github.com/squidfunk/mkdocs-material/blob/ac3315fd3e25b9449306d0dac6cd535d4d32d9ea/src/plugins/privacy/plugin.py#L470

When using named groups here this should work consistently:

```sh
(.venv) z003wwtz@framework:~/tmp/9.5.43-privacy-plugin-bug$ mkdocs build
INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: /home/z003wwtz/tmp/9.5.43-privacy-plugin-bug/public
INFO    -  Downloading external file: https://unpkg.com/mermaid@11/dist/mermaid.min.js
INFO    -  Downloading external file:
           https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TjASc3CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TjASc-CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TjASc2CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TjASc5CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TjASc1CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TjASc0CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TjASc6CsQ.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOkCnqEu92Fr1Mu51xFIzIFKw.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOkCnqEu92Fr1Mu51xMIzIFKw.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOkCnqEu92Fr1Mu51xEIzIFKw.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOkCnqEu92Fr1Mu51xLIzIFKw.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOkCnqEu92Fr1Mu51xGIzIFKw.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOkCnqEu92Fr1Mu51xHIzIFKw.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOkCnqEu92Fr1Mu51xIIzI.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TzBic3CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TzBic-CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TzBic2CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TzBic5CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TzBic1CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TzBic0CsTKlA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOjCnqEu92Fr1Mu51TzBic6CsQ.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmSU5fCRc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmSU5fABc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmSU5fCBc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmSU5fBxc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmSU5fChc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmSU5fBBc4.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmSU5fCxc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOmCnqEu92Fr1Mu72xKOzY.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOmCnqEu92Fr1Mu5mxKOzY.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOmCnqEu92Fr1Mu7mxKOzY.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOmCnqEu92Fr1Mu4WxKOzY.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOmCnqEu92Fr1Mu7WxKOzY.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOmCnqEu92Fr1Mu7GxKOzY.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOmCnqEu92Fr1Mu4mxK.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmWUlfCRc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmWUlfABc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmWUlfCBc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmWUlfBxc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmWUlfCxc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmWUlfChc4EsA.woff2
INFO    -  Downloading external file: https://fonts.gstatic.com/s/roboto/v32/KFOlCnqEu92Fr1MmWUlfBBc4.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEluUlYIw.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEn-UlYIw.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEmOUlYIw.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtElOUlYIw.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEleUlYIw.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEm-Ul.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSV0mf0h.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSx0mf0h.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSt0mf0h.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSd0mf0h.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSZ0mf0h.woff2
INFO    -  Downloading external file:
           https://fonts.gstatic.com/s/robotomono/v23/L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSh0mQ.woff2
INFO    -  Documentation built in 1.87 seconds
(.venv) z003wwtz@framework:~/tmp/9.5.43-privacy-plugin-bug$ tree public/assets/external/
public/assets/external/
├── fonts.googleapis.com
│   └── css.49ea35f2.css
├── fonts.gstatic.com
│   └── s
│       ├── roboto
│       │   └── v32
│       │       ├── KFOjCnqEu92Fr1Mu51TjASc0CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TjASc1CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TjASc2CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TjASc3CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TjASc5CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TjASc6CsQ.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TjASc-CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TzBic0CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TzBic1CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TzBic2CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TzBic3CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TzBic5CsTKlA.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TzBic6CsQ.woff2
│       │       ├── KFOjCnqEu92Fr1Mu51TzBic-CsTKlA.woff2
│       │       ├── KFOkCnqEu92Fr1Mu51xEIzIFKw.woff2
│       │       ├── KFOkCnqEu92Fr1Mu51xFIzIFKw.woff2
│       │       ├── KFOkCnqEu92Fr1Mu51xGIzIFKw.woff2
│       │       ├── KFOkCnqEu92Fr1Mu51xHIzIFKw.woff2
│       │       ├── KFOkCnqEu92Fr1Mu51xIIzI.woff2
│       │       ├── KFOkCnqEu92Fr1Mu51xLIzIFKw.woff2
│       │       ├── KFOkCnqEu92Fr1Mu51xMIzIFKw.woff2
│       │       ├── KFOlCnqEu92Fr1MmSU5fABc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmSU5fBBc4.woff2
│       │       ├── KFOlCnqEu92Fr1MmSU5fBxc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmSU5fCBc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmSU5fChc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmSU5fCRc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmSU5fCxc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmWUlfABc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmWUlfBBc4.woff2
│       │       ├── KFOlCnqEu92Fr1MmWUlfBxc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmWUlfCBc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmWUlfChc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmWUlfCRc4EsA.woff2
│       │       ├── KFOlCnqEu92Fr1MmWUlfCxc4EsA.woff2
│       │       ├── KFOmCnqEu92Fr1Mu4mxK.woff2
│       │       ├── KFOmCnqEu92Fr1Mu4WxKOzY.woff2
│       │       ├── KFOmCnqEu92Fr1Mu5mxKOzY.woff2
│       │       ├── KFOmCnqEu92Fr1Mu72xKOzY.woff2
│       │       ├── KFOmCnqEu92Fr1Mu7GxKOzY.woff2
│       │       ├── KFOmCnqEu92Fr1Mu7mxKOzY.woff2
│       │       └── KFOmCnqEu92Fr1Mu7WxKOzY.woff2
│       └── robotomono
│           └── v23
│               ├── L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEleUlYIw.woff2
│               ├── L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtElOUlYIw.woff2
│               ├── L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEluUlYIw.woff2
│               ├── L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEmOUlYIw.woff2
│               ├── L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEm-Ul.woff2
│               ├── L0xdDF4xlVMF-BfR8bXMIjhOsXG-q2oeuFoqFrlnAIe2Imhk1T8rbociImtEn-UlYIw.woff2
│               ├── L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSd0mf0h.woff2
│               ├── L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSh0mQ.woff2
│               ├── L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSt0mf0h.woff2
│               ├── L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSV0mf0h.woff2
│               ├── L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSx0mf0h.woff2
│               └── L0xTDF4xlVMF-BfR8bXMIhJHg45mwgGEFl0_3vrtSM1J-gEPT5Ese6hmHSZ0mf0h.woff2
└── unpkg.com
    └── mermaid@11
        └── dist
            └── mermaid.min.js

10 directories, 56 files
```

Just need to check that it is applied consistently when using `value = match.group("url")` Should have added some tests in the PR :sweat_smile: 

@squidfunk is this something you're already looking into or should I start a PR?

**Comment by squidfunk** [comment]
Thanks for the very fast fix @nejch! Keeping open until released.

**Comment by squidfunk** [comment]
Released as part of 9.5.44. Thanks again, @nejch!

**Comment by dr-br** [comment]
Excellent response time!
Thanks and Dankeschön :)
